### PR TITLE
Add onErrorComplete operator and rework onErrorReturn

### PR DESCRIPTION
**Pull request by simonbasle**
This PR introduces a new implementation that optimizes the onErrorReturn case to avoid going through a generated `Publisher`.

Additionally, this implementation can easily accept a `null` fallback value, which covers a common case that has since now been covered by `onErrorResume(e -> Mono.empty())`.

We take that as an opportunity to introduce a convenient `onErrorComplete` operator for that later case.

### CODE REVIEW

**Comment by simonbasle** [comment]
(rebased on latest 3.4.x and squashed the numerous commits into a single one)

**Comment by simonbasle** [comment]
TODO:
 - [ ] marble diagram for `onErrorComplete`
 - [ ] validate the name
 - [ ] reference it in the refguide / operator choice matrix

File: reactor-core/src/main/java/reactor/core/publisher/FluxOnErrorReturn.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+			//we try to turn that 0 into a marker for pending fallback but if we fail we might want to propagate fallback ourselves
+			if (!REQUESTED.compareAndSet(this, 0, STATE_PENDING_FALLBACK)) {
+				//re-check the request:
+				r = this.requested;
+				if (r > 0) {
+					REQUESTED.set(this, STATE_TERMINATED);
+					actual.onNext(this.fallbackValue);
+					actual.onComplete();
+				}
+			} // else next request will take care of delivering
```
Comment (chemicL) [inline reactor-core/src/main/java/reactor/core/publisher/FluxOnErrorReturn.java:215]: If `r <= 0` in the inner condition above, the only option is that it was cancelled, correct? Perhaps worth adding a comment about it?

**Review by chemicL — APPROVED** [review]

File: reactor-core/src/main/java/reactor/core/publisher/FluxOnErrorReturn.java (hunk 1 above)
Comment (simonbasle) [reply reactor-core/src/main/java/reactor/core/publisher/FluxOnErrorReturn.java:215]: good idea !

**Pull request merged**